new-lamassu-admin/src/pages/Dashboard/SystemPerformance/SystemPerformance.js

```diff
@@ -0,0 +1,229 @@
+import { useQuery } from '@apollo/react-hooks'
+import Grid from '@material-ui/core/Grid'
+import { makeStyles } from '@material-ui/core/styles'
+import BigNumber from 'bignumber.js'
+import gql from 'graphql-tag'
+import moment from 'moment'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+
+import { Label2 } from 'src/components/typography/index'
+import { ReactComponent as TriangleDown } from 'src/styling/icons/arrow/triangle_down.svg'
+import { ReactComponent as TriangleUp } from 'src/styling/icons/arrow/triangle_up.svg'
+import { fromNamespace } from 'src/utils/config'
+
+import PercentageChart from './Graphs/PercentageChart'
+import LineChart from './Graphs/RefLineChart'
+import Scatterplot from './Graphs/RefScatterplot'
+import InfoWithLabel from './InfoWithLabel'
+import Nav from './Nav'
+import styles from './SystemPerformance.styles'
+
+BigNumber.config({ ROUNDING_MODE: BigNumber.ROUND_HALF_UP })
+
+const getFiats = R.map(R.prop('fiat'))
+const useStyles = makeStyles(styles)
+const mapToFee = R.map(R.prop('cashInFee'))
+
+const getDateSecondsAgo = (seconds = 0, startDate = null) => {
+  const date = startDate ? moment(startDate) : moment()
+  return date.subtract(seconds, 'second')
+}
+
+const ranges = {
+  Day: {
+    left: getDateSecondsAgo(2 * 24 * 3600, moment()),
+    right: getDateSecondsAgo(24 * 3600, moment())
+  },
+  Week: {
+    left: getDateSecondsAgo(14 * 24 * 3600, moment()),
+    right: getDateSecondsAgo(7 * 24 * 3600, moment())
+  },
+  Month: {
+    left: getDateSecondsAgo(60 * 24 * 3600, moment()),
+    right: getDateSecondsAgo(30 * 24 * 3600, moment())
+  }
+}
+
+const GET_DATA = gql`
+  query getData {
+    transactions {
+      fiatCode
+      fiat
+      cashInFee
+      commissionPercentage
+      created
+      txClass
+      error
+    }
+    fiatRates {
+      code
+      name
+      rate
+    }
+    config
+  }
+`
+
+const SystemPerformance = () => {
+  const classes = useStyles()
+  const [selectedRange, setSelectedRange] = useState('Day')
+  const { data, loading } = useQuery(GET_DATA)
+  const fiatLocale = fromNamespace('locale')(data?.config).fiatCurrency
+
+  const isInRangeAndNoError = getLastTimePeriod => t => {
+    if (getLastTimePeriod) {
+      return (
+        t.error === null &&
+        moment(t.created).isBetween(
+          ranges[selectedRange].left,
+          ranges[selectedRange].right
+        )
+      )
+    }
+    return (
+      t.error === null &&
+      moment(t.created).isBetween(ranges[selectedRange].right, moment())
+    )
```

Comment:
Seems like if t.error !== null we're returning false. So you can add this if atop the function

```
if (t.error !== null) return false

if (!getLastTimePeriod) {
  return moment(t.created).isBetween(ranges[selectedRange].right, moment())
}

return  moment(t.created).isBetween(
          ranges[selectedRange].left,
          ranges[selectedRange].right
        )
```